src/data/finset/basic.lean

```diff
@@ -2694,6 +2694,11 @@ lemma exists_list_nodup_eq [decidable_eq α] (s : finset α) :
   ∃ (l : list α), l.nodup ∧ l.to_finset = s :=
 ⟨s.to_list, s.nodup_to_list, s.to_list_to_finset⟩
 
+lemma to_list_insert [decidable_eq α] {a : α} {s : finset α} (h : a ∉ s) :
+  (insert a s).to_list ~ a :: s.to_list :=
+(list.perm_ext (nodup_to_list _) (by simp [h, nodup_to_list s])).2 $
+  λ x, by simp only [list.mem_cons_iff, iff_self, finset.mem_to_list, finset.mem_insert]
```

Comment:
Can you add `to_list_cons` as well for good measure? It might even be possible to prove one in terms of the other.

Reply:
done